vllm/model_executor/models/llama.py

```diff
@@ -402,3 +414,28 @@ def load_weights(self,
                 weight_loader = getattr(param, "weight_loader",
                                         default_weight_loader)
                 weight_loader(param, loaded_weight)
+
+    # Should not be called unless the KV cache dtype is FP8 on ROCm (AMD GPU)
```

Comment:
in that case please use assert to confirm the invariant.

Reply:
I've removed the comment, it was meant to be more informative ("FYI, this function is for scaled KV cache, which is currently enabled on ROCm only") rather than bad behavior we should guard against. In particular, the scaling factor is only used when the KV cache dtype is FP8 and on ROCm, so calling this function in other settings has no observable side effects. So there's no need for an assert here.

More broadly, the current design largely decouples the KV cache implementation from the model implementation (which makes sense, as KV caches are not theoretically necessary). IMO, guarding against potential misuse (which is side effect free anyway) isn't a strong enough reason to newly endow the model with the ability to introspect KV cache details.